src/selectors/entities/channels.ts

```diff
@@ -892,3 +892,7 @@ export function isManuallyUnread(state: GlobalState, channelId?: string): boolea
 
     return Boolean(state.entities.channels.manuallyUnread[channelId]);
 }
+
+export function getChannelModerations(state: GlobalState, channelId: string): Array<ChannelModeration> {
+    return state.entities.channels.channelModerations[channelId] || [];
```

Comment:
This returns a new array each time it's called which can cause some bad behaviour when this is used in `mapStateToProps`. Can you either allow this to return undefined or make this into a selector? If made into a selector, it needs to be a selector factory (like `makeGetChannelModerations` that returns an instance of `createSelector`) since it has a parameter that needs to be memoized.

Reply:
@hahmadia I'm going to opt for making this return an `undefined` if there is no `channelModerations[channelId]` found from the selector - let me know if this causes any issues for you and if you'd prefer to go the selector factory route